service-worker/service-worker-generator.js

```diff
@@ -45,7 +47,10 @@ function generateRuntimeCaching(runtimeCaching) {
   }, '')
 }
 
-const apiURLPrefix = `${config.API_PROTOCOL}://${config.API_HOST}(:${config.API_PORT})?`
+const releaseBranch = config.releaseBranch
+const origin = url.parse(requestOrigins.forClientSideRendering[releaseBranch])
+
+const apiURLPrefix = `${origin.protocol}://${origin.hostname}(:${config.port})?`
```

Comment:
Since `requestOrigins.forClientSideRendering[releaseBranch].api` is a url origin,
how about 
```
const apiURLPrefix = `${requestOrigins.forClientSideRendering[releaseBranch].api}?`
```
?

Reply:
I parse the url in order to get `port` (it's optional in our matching rule)

Reply:
Got it. 
I kind of forget why port is optional.
However, remember add `.api` after `forClientSideRender[releaseBranch]` .